### DISCUSSION TITLE
API for accessing tasty trees

### DISCUSSION THREAD

**Discussion by civp** [scalacenter/tasty-query#54] [Q&A] [answered]
In my program, I want to get the tree of a single class and do some analyses on it. But I don't know if I'm donig it correctly. I have to modify the access modifiers of `trees` in `TastyQuery` and `TastyTrees` as a temporary workaround. Please correct me if I did it wrong.

```scala
val reader = new ProjectReader
val query = reader.read(className)
val tree = query.trees.trees.head
checkTree(tree)
```

**Comment by bishabosha** [comment, marked as answer]
I think we will probably build a higher level api over trees, e.g. a "TastyFile" or something, so this trees field may not exist in the future
